### PR TITLE
Publish Nuget package with Jenkins pipeline

### PR DESCRIPTION
**Pull request by rs--**

### CODE REVIEW

File: .jenkins/pipelines/OpenEnclave/releases/publish/Jenkinsfile [hunk 1]
```diff
@@ -16,59 +16,42 @@ library "OpenEnclaveJenkinsLibrary@${params.OECI_LIB_VERSION}"
...
 def ubuntuPublishRelease(String release_version, String os_release) {
 
     helpers.WaitForAptLock()
     withCredentials([file(credentialsId: 'azure-repoapi-client', variable: 'repoapiclient'),
-                     string(credentialsId: 'msft-apt-repo-pem', variable: 'repoclientpem'),
-                     string(credentialsId: 'ReleaseAADClientId', variable: 'AADClientId'),
-                     string(credentialsId: 'ReleaseAADResource', variable: 'AADResource'),
-                     string(credentialsId: 'ReleaseAADTenant', variable: 'AADTenant'),
-                     string(credentialsId: "MicrosoftAptRepositoryUbuntu${os_release}", variable: "Ubuntu${os_release}repoid")]) {
+                     file(credentialsId: "ubuntu-${os_release}-repo-config", variable: 'aptconfig'),
```
Comment (rs--) [inline .jenkins/pipelines/OpenEnclave/releases/publish/Jenkinsfile:27]: For the time being, this file is uploaded directly to Jenkins

File: .jenkins/pipelines/OpenEnclave/releases/publish/Jenkinsfile [hunk 2]
```diff
@@ -16,59 +16,42 @@ library "OpenEnclaveJenkinsLibrary@${params.OECI_LIB_VERSION}"
...
+                     string(credentialsId: 'msft-apt-repo-pem', variable: 'repoclientpem')]) {
         sh '''#!/bin/bash
             set +x
-            echo Ubuntu${os_release}
+            echo Ubuntu $os_release
```
Comment (CyanDevs) [inline .jenkins/pipelines/OpenEnclave/releases/publish/Jenkinsfile:31]: What is the purpose of removing the braces here and throughout this file?

File: .jenkins/pipelines/OpenEnclave/releases/publish/Jenkinsfile [hunk 3]
```diff
@@ -16,59 +16,42 @@ library "OpenEnclaveJenkinsLibrary@${params.OECI_LIB_VERSION}"
...
-    "AADClientId": "${AADClientId}",
-    "AADClientCertificate": "${WORKSPACE}/repoclient.pem",
-    "AADResource": "${AADResource}",
-    "AADTenant": "${AADTenant}",
-    "AADAuthorityUrl": "https://login.microsoftonline.com",
-    "repositoryId": "Ubuntu${os_release}repoid"
-}
-EOF
+            cp $aptconfig $WORKSPACE/apt-config.json
+            chmod 777 $WORKSPACE/apt-config.json
```
Comment (CyanDevs) [inline .jenkins/pipelines/OpenEnclave/releases/publish/Jenkinsfile:49]: Please keep the config to 600 so it is not world readable

File: .jenkins/pipelines/OpenEnclave/releases/publish/Jenkinsfile [hunk 4]
```diff
@@ -77,15 +60,46 @@ EOF
                 helpers.releaseDownloadLinux(release_version, "open-enclave-hostverify", 'GitHub', 'Ubuntu', os_release)
                 sh """#!/bin/bash
                     echo "Publishing OpenEnclave ${release_version}"
-                    repoclient -c ${WORKSPACE}/config-Ubuntu${os_release}.json package add open-enclave"
-                    repoclient -c ${WORKSPACE}/config-Ubuntu${os_release}.json package add open-enclave-hostverify"
+                    mv "0.17.5/Ubuntu ${os_release}/Ubuntu_${os_release}_open-enclave_${release_version}_amd64.deb" open-enclave_${release_version}_amd64.deb
```
Comment (CyanDevs) [inline .jenkins/pipelines/OpenEnclave/releases/publish/Jenkinsfile:63]: Don't hardcode the version here. Use `${release_version}`

File: .jenkins/pipelines/OpenEnclave/releases/publish/Jenkinsfile (hunk 1 above)
Comment (CyanDevs) [reply .jenkins/pipelines/OpenEnclave/releases/publish/Jenkinsfile:27]: Could you explain the benefit of doing this via generating it in the job?

File: .jenkins/pipelines/OpenEnclave/releases/publish/Jenkinsfile [hunk 5]
```diff
@@ -77,15 +60,46 @@ EOF
...
+                @echo off
+                wmic os get caption | find /v "Caption"
+                """,
+            returnStdout: true
+        ).trim()
+
+        helpers.releaseDownloadWindows(release_version, 'open-enclave', 'GitHub', windows_version)
+        helpers.releaseDownloadWindows(release_version, 'open-enclave.OEHOSTVERIFY', 'GitHub', windows_version)
+        script {
+            if (params.PUBLISH_RELEASE) {
```
Comment (CyanDevs) [inline .jenkins/pipelines/OpenEnclave/releases/publish/Jenkinsfile:93]: Maybe better to have it so the entire function wouldn't run anything if params.PUBLISH_RELEASE is false? Going a step further, why call this function if it is false?

File: .jenkins/pipelines/OpenEnclave/releases/publish/Jenkinsfile [hunk 6]
```diff
@@ -16,59 +16,42 @@ library "OpenEnclaveJenkinsLibrary@${params.OECI_LIB_VERSION}"
...
             # Install repoclient.pem
-            if [[ -f ${WORKSPACE}/repoclient.pem ]]; then
-                rm ${WORKSPACE}/repoclient.pem
+            if [[ -f $WORKSPACE/repoclient.pem ]]; then
+                rm $WORKSPACE/repoclient.pem
             fi
             touch ~/.ssh/repoclient.pem
             chmod 600 ~/.ssh/repoclient.pem
-            echo ${repoclientpem} | awk 'NF {gsub(/- /, "-\\n"); gsub(/ -/, "\\n-"); printf \$0;}' &>> ${WORKSPACE}/repoclient.pem
+            echo "$repoclientpem" >> $WORKSPACE/repoclient.pem
```
Comment (CyanDevs) [inline .jenkins/pipelines/OpenEnclave/releases/publish/Jenkinsfile:45]: This is nice as we can avoid the awk. Though are you sure this works? Last time I tried this, the pem was considered invalid due to the bad formatting.

**Review by CyanDevs — APPROVED** [review]
Please squash before merging. Thanks!

**Pull request closed without merging**